src/Education.UnitTests/RuleHelpXamlTranslatorTests.cs

```diff
@@ -147,11 +149,160 @@ public void TranslateHtmlToXaml_Span_AddsSpan()
             result.Should().Be(expectedText);
         }
 
-        private static IRuleHelpXamlTranslator CreateTestSubject(IXamlWriterFactory xamlWriterFactory = null)
+        [TestMethod]
+        public void TranslateHtmlToXaml_DataDiffExists_HighlightsCode()
+        {
+            IRuleHelpXamlTranslator testSubject = CreateTestSubject();
+
+            var compliantText = @"Same text 1
+diff 1
+same text 2";
+            var nonCompliantText = @"Same text 1
+diff 2
+same text 2";
+
+            var htmlText = $"<pre data-diff-type=\"compliant\" data-diff-id=\"1\">{compliantText}</pre>\n<pre data-diff-type =\"noncompliant\" data-diff-id=\"1\">{nonCompliantText}</pre>";
+
+            var expectedText = @"<Section xml:space=""preserve"" Style=""{DynamicResource Pre_Section}"">
+  <Paragraph>Same text 1
+<Span Style=""{DynamicResource Compliant_Diff}"">diff 1</Span>
+same text 2</Paragraph>
+</Section>
+<Section xml:space=""preserve"" Style=""{DynamicResource Pre_Section}""><Paragraph>Same text 1
+<Span Style=""{DynamicResource NonCompliant_Diff}"">diff 2</Span>
+same text 2</Paragraph></Section>";
+
+            var result = testSubject.TranslateHtmlToXaml(htmlText);
+
+            result.Replace("\r\n", "\n").Should().Be(expectedText.Replace("\r\n", "\n"));
```

Comment:
There were many irregularities of break ends so I have decided to ignore them in comparison. This wouldn't affect generated xaml.